### DISCUSSION TITLE
Updating helm charts with renovate from non-standard sources (like helm-gcs)

### DISCUSSION THREAD

**Discussion by leevs** [renovatebot/renovate#19927] [Request Help]
I created the following [featue-request to support helm chart dep. updates with the use of helm-gcs](https://github.com/renovatebot/renovate/issues/19902). 
This got me thinking, while [helm-gcs](https://github.com/hayorov/helm-gcs) is a plugin for helm to support yet another storage, it is a good usecase for a cost-effective one (easy to manage with lifecycles also). 

Knowing people will create more of these extra integrations/ways, should this be supported and how? 
Just thinking if the system (where renovate is running) supports a local lookup of helm charts & versions, that would give the ownership to a well-configured system (instead of renovate trying to understand multiple sources). 

I might be completely wrong, which is why I look for help/feedback.

**Comment by rarkins** [comment]
It sounds like GCS is being used for file storage here, but the format is helm's? In other words:
- Helm may have many different types of file storage endpoints (S3, GCS, etc)
- Each storage endpoint, e.g. S3 or GCS, may have many types of file formats which are stored there (e.g. helm, others).

From a datasource point of view, I think arrange around the file _format_ and not the storage type. Storage types could be put in a generic location so that they can be reused between datasources. In other words this helm-gcs support likely belongs in the `helm` datasource, but we'll need a way for the datasource to know what type of storage type it is, e.g.
- Due to well-known registryUrl https://storage.googleapis.com, or
- Using new `gs://` protocol

**Reply by leevs** [reply]
it will be the `gs://` protocol:

```bash
helm repo list
tools	gs://xxxxxx/charts
```
This is also the same in the `Chart.yaml`

```yaml
dependencies:
- name: dep-chart-a
  repository: gs://xxxxxxxxxxx/charts
  version: 1.1.1
```

**Reply by rarkins** [reply]
I believe right now Renovate enforces that repository URLs must start with https:// so there may be some miscellaneous validation code needing updates too

**Reply by leevs** [reply]
That would explain the reason why I get the following `ERR_UNSUPPORTED_PROTOCOL` during a run with `DEBUG` on. 

```
DEBUG: Datasource connection error (repository=xxx/xxxxx)
       "datasource": "helm",
       "packageName": "<chartName>",
       "url": undefined,
       "errCode": "ERR_UNSUPPORTED_PROTOCOL"
```

**Comment by leevs** [comment]
After a modification of `gs://` to `https://storage.googleapis.com/` I  can see it's working. It's able to connect. 
Not sure how we take this further. Is there a possibility to move this to a supported protocol or should it be adjusted in the documentation?

**Reply by rarkins** [reply]
Perhaps the manager could rewrite gs:// to https:// ? Docs would help too
